src/source_sampling.cpp

```diff
@@ -394,3 +590,47 @@ int pyne::AliasTable::sample_pdf(double rand1, double rand2) {
   int i = (int) n * rand1;
   return rand2 < prob[i] ? i : alias[i];
 }
+
+pyne::SourceParticle::SourceParticle(double x, double y, double z,
```

Comment:
Nice work with the addition of this class! I think this benefits not only your changes but the entire source sampling routine.

I do wonder if maybe the variable names could be more simple here. Rather than `xxx`, `yyy`, `zzz` why not `x`, `y`, `z`? Maybe there is a convention here I'm not aware of...

Reply:
`x`, `y`, `z` are used as the input parameters of the constructor of the class `SourceParticle` and also as the input parameters of `particle_birth`. If the `x`, `y`, `z` are then used as the names of the attributes of the source particle, the `SourceParticle` constructor will be something like below:
```
pyne::SourceParticle::SourceParticle(double x, double y, double z,
                                     double e, double w, int c) {
    x = x;
    y = y;
    z = z;
    e = e;
    w = w;
    c = c;
}
```
In other words, names of the attributes and the input arguments will be the same. It may lead to confusion of the attributes and arguments. Therefore, I think it's better to use different names to differentiate the attributes from the arguments for better readability. 
So another set of variable names `xxx`, `yyy`, `zzz`, `erg`, `wgt` are used. They are the variable names used in `MCNP`.

Reply:
This is true. And I think it is important to differentiate the argument variables from the attributes of the class. The arguments to the constructor are temporary, however, and won't be seen even when calling the class constructor. The attributes of the class are more permanent and in my opinion should probably have priority in for a clear naming scheme over the constructor arguments.

I'm thinking that even something like this should suffice: 

```
pyne::SourceParticle::SourceParticle(double _x, double _y, double _z,
                                     double _e, double _w, int _c) {
    x = _x;
    y = _y;
    z = _z;
    e = _e;
    w = _w;
    c = _c;
}
```

Separate from this suggestion is the naming of methods on the class. Even though the attribute may be named `xxx` or `x` the method `get_source_xxx` can still be `get_src_x` which makes sense regardless of whether or not you know the MCNP conventions for these variable names.

These are small things, but they may become important in the long run :)

Reply:
I found that I can use `this` to differentiate the attributes from the parameters. So the following code may be a better way.  
```
pyne::SourceParticle::SourceParticle(double x, double y, double z,
                                     double e, double w, int c) {
    this->x = x;
    this->y = y;
    this->z = z;
    this->e = e;
    this->w = w;
    this->c = c;
}
```
And I changed the name of `xxx` to `x`, changed the name of `get_src_xxx` to `get_src_x` according to your suggestion.